cpp/src/arrow/flight/client.h

```diff
@@ -301,6 +301,17 @@ class ARROW_FLIGHT_EXPORT FlightClient {
     return GetFlightInfo({}, descriptor);
   }
 
+  /// \brief Asynchronous GetFlightInfo.
+  /// \param[in] options Per-RPC options
+  /// \param[in] descriptor the dataset request
+  /// \param[in] listener Callbacks for response and RPC completion
+  void GetFlightInfo(const FlightCallOptions& options, const FlightDescriptor& descriptor,
+                     std::shared_ptr<AsyncListener<FlightInfo>> listener);
+  void GetFlightInfo(const FlightDescriptor& descriptor,
```

Comment:
Hmm, I was thinking the extra few characters weren't quite necessary, but you're right in that we name everything else async with an explicit Async in Arrow.

Reply:
I don't feel very strongly either way.  Some languages have this convention (e.g. C#) and others do not (e.g. Javascript).

Reply:
However, it is necessary, when there are both synchronous and asynchronous versions of a method (since overloads cannot differ only on return type).  This is probably my original motivation for the naming scheme.